lib/gems/pending/VMwareWebService/MiqVimDataStore.rb

```diff
@@ -287,7 +286,14 @@ def get_file_content(filepath)
     /(\d*)\.(\d*)/ =~ @invObj.about['version']
     raise "get_file_content not supported on [#{@invObj.about['fullName']}]" if $1.to_i < 3 || ($1.to_i == 3 && $2.to_i < 5)
 
-    fileUrl = "https://#{@invObj.server}/folder/#{MIQEncode.base64Encode(filepath)}?dsName=#{MIQEncode.base64Encode(@name)}"
+    uri_parts = {
+      :scheme => "https",
+      :host   => @invObj.server,
+      :path   => "/folder/#{filepath}",
+      :query  => "dsName=#{@name}"
+    }
+
+    fileUrl = URI::Generic.build(uri_parts).to_s
```

Comment:
Building the URI like this is much nicer, but I think we still need to `URI.encode(data, /[^#{URI::PATTERN::UNRESERVED}]/)`.
```ruby
irb(main):001:0> MIQEncode.base64Encode("a%b&c")
=> "a%25b%26c"
irb(main):002:0> URI::Generic.build(:scheme => "https", :host => "aaa", :path => "/zzz/", :query => "dsname=a%b&c").to_s
=> "https://aaa/zzz/?dsname=a%b&c"
```
Also, do you know what special characters can be in the `filepath`?
```ruby
irb(main):003:0> URI::Generic.build(:scheme => "https", :host => "aaa", :path => "/zzz/x%y^z", :query => "dsname=a%b&c").to_s
URI::InvalidComponentError: bad component(expected absolute path component): /zzz/x%y^z
	from /home/bdunne/.rubies/ruby-2.3.3/lib/ruby/2.3.0/uri/generic.rb:769:in `check_path'
	from /home/bdunne/.rubies/ruby-2.3.3/lib/ruby/2.3.0/uri/generic.rb:817:in `path='
	from /home/bdunne/.rubies/ruby-2.3.3/lib/ruby/2.3.0/uri/generic.rb:191:in `initialize'
	from /home/bdunne/.rubies/ruby-2.3.3/lib/ruby/2.3.0/uri/generic.rb:136:in `new'
	from /home/bdunne/.rubies/ruby-2.3.3/lib/ruby/2.3.0/uri/generic.rb:136:in `build'
	from (irb):4
```